woocommerce/woocommerce-android
# [Store Onboarding] Show an explanation screen before opening the Payment's task WebView

Closes: #9731 


### Description
This PR simply adds a pre-setup screen to the regular `Payments` task in the onboarding, it reuses the same screen as WooPayments, but with different text and URLs.

Since we are reusing the screen for both flows now, the name that included `Terms` doesn't fit anymore, so I renamed the different classes to make their role clearer, for this reason, please review 2a3c6db separately as it just renames classes, and might be a noise when reviewing the other files.

### Testing instructions
1. Create a new selfhosted store (Jetpack or not, it doesn't matter, just make sure it doesn't have WooPayments)
2. Open the app then sign-in.
3. Click on the "Get Paid" task.
4. Confirm the pre-setup screen is shown.
5. Click on "Learn More" and confirm it opens a support page.
6. Go back then click on "Continue" and confirm the setup begins.

### Images/gif
<img width=400 src="https://github.com/woocommerce/woocommerce-android/assets/1657201/f3911b9e-29b9-4ef8-a06d-ef705e7050d8"/>
